metrics/sources/kubelet/kubelet.go

```diff
@@ -235,15 +235,19 @@ metricloop:
 
 func (this *kubeletMetricsSource) ScrapeMetrics(start, end time.Time) *DataBatch {
 	containers, err := this.scrapeKubelet(this.kubeletClient, this.host, start, end)
-	if err != nil {
-		glog.Errorf("error while getting containers from Kubelet %s: %v", this.host, err)
-	}
-	glog.V(2).Infof("successfully obtained stats from %s for %v containers", this.host, len(containers))
 
 	result := &DataBatch{
 		Timestamp:  end,
 		MetricSets: map[string]*MetricSet{},
 	}
+
```

Comment:
we shouldn't be returning a result at all if there's an error.  Change the signature of the method to also return an error (you might also have to change the signature of the interface for metric sources).